crates/typst-library/src/layout/flow.rs

```diff
@@ -436,6 +440,24 @@ impl<'a> FlowLayouter<'a> {
 
     /// Finish the frame for one region.
     fn finish_region(&mut self, vt: &mut Vt) -> SourceResult<()> {
+        if let Some(items) = self.last_frame_only_out_of_flow.take() {
+            self.items.splice(0..0, items);
+            self.finished.pop();
+            self.regions.next();
+            self.initial = self.regions.size;
+            self.has_footnotes = false;
+        }
+
+        if !self.items.is_empty()
+            && self.items.iter().all(|item| match item {
+                FlowItem::Placed { float: false, .. } => true,
+                FlowItem::Frame { frame, .. } => frame.size().is_zero(),
```

Comment:
Not sure what you mean here. The condition is `!self.items.is_empty() && …`. So an empty frame is never considered out-of-flow.

Reply:
Sorry, I meant "I don't think a frame of size zero is necessarily out of flow". It might be size zero because it contains just metadata, but it might as well contain a box of height 0.